qcodes/instrument_drivers/Keysight/N9030B.py

```diff
@@ -0,0 +1,542 @@
+        """
+        Sets up the Log Plot measurement sweep for Phase Noise Mode.
+        """
+        if "LPLot" in self._available_meas():
+            self.measurement("LPLot")
+        else:
+            raise RuntimeError("Log Plot measurement is not available on your "
+                               "Keysight N9030B instrument with Phase Noise "
+                               "mode.")
+
+        self.start_offset(start_offset)
+        self.stop_offset(stop_offset)
+        self.npts(npts)
+
+    def autotune(self) -> None:
+        """
+        Autotunes frequency
+        """
+        self.write(":SENSe:FREQuency:CARRier:SEARch")
+
+
+class N9030B(VisaInstrument):
+    """
+    Driver for Keysight N9030B PXA signal analyzer.
+    """
+
+    CHANNEL_CLASS_1 = SpectrumAnalyzer
+    CHANNEL_CLASS_2 = PhaseNoise
+
+    def __init__(self, name: str, address: str, **kwargs: Any) -> None:
+        super().__init__(name, address, terminator='\n', **kwargs)
+
+        self._min_freq: float
+        self._max_freq: float
+        self._additional_wait: float = 1
+
+        self.add_parameter(
+            name="mode",
+            get_cmd=":INSTrument:SELect?",
+            set_cmd=":INSTrument:SELect {}",
+            vals=Enum(*self._available_modes()),
+            docstring="Allows setting of different modes present and licensed "
+                      "for the instrument."
+        )
+
+        self.add_parameter(
+            name="measurement",
+            get_cmd=":CONFigure?",
+            set_cmd=":CONFigure:{}",
+            vals=Enum(*self._available_meas()),
+            docstring="Sets measurement type from among the available "
+                      "measurement types."
+        )
+
+        self.add_parameter(
+            name="cont_meas",
+            initial_value=False,
+            get_cmd=":INITiate:CONTinuous?",
+            set_cmd=self._enable_cont_meas,
+            val_mapping=create_on_off_val_mapping(on_val="ON", off_val="OFF"),
+            docstring="Enables or disables continuous measurement."
+        )
+
+        self.add_parameter(
+            name="format",
+            get_cmd=":FORMat:TRACe:DATA?",
+            set_cmd=":FORMat:TRACe:DATA {}",
+            val_mapping={
+                "ascii": "ASCii",
+                "int32": "INTeger,32",
+                "real32": "REAL,32",
+                "real64": "REAL,64"
+            },
+            docstring="Sets up format of data received"
+        )
+
+        if "SA" in self._available_modes():
+            sa_mode = ChannelList(
+                self, "SA", self.CHANNEL_CLASS_1, snapshotable=True
+            )
+            self.add_submodule("SA", sa_mode)
+        else:
+            self.log.info("Spectrum Analyzer mode is not available on this "
+                          "instrument.")
+
+        if "PNOISE" in self._available_modes():
+            pnoise_mode = ChannelList(
+                self, "PNoise", self.CHANNEL_CLASS_2, snapshotable=True
```

Comment:
typically we use lower case naming for sub modules. Suggest using either pn or phase_noise.